src/backend/platform_info.rs

```diff
@@ -170,5 +182,74 @@ pub(super) async fn run_platform_task<'s>(sdk: &Sdk, task: PlatformInfoTask) ->
                 },
             }
         }
+        PlatformInfoTask::FetchNodeStatuses => {
+            let address_list = match sdk.address_list() {
+                Ok(list) => list,
+                Err(e) => {
+                    return BackendEvent::TaskCompleted {
+                        task: Task::PlatformInfo(task),
+                        execution_result: Err(format!(
+                            "Failed to fetch DapiClient address list: {e}"
+                        )),
+                    }
+                }
+            };
+
+            let settings = RequestSettings {
+                connect_timeout: Some(Duration::from_secs(10)),
+                timeout: Some(Duration::from_secs(10)),
+                retries: None,
+                ban_failed_address: Some(false),
+            };
+
+            let mut handles = Vec::new();
+
+            for address in address_list {
+                let sdk = sdk.clone();
+                let handle = task::spawn(async move {
+                    let node = EvoNode::new(address);
+                    match EvoNodeStatus::fetch_unproved_with_settings(&sdk, node.clone(), settings)
+                        .await
+                    {
+                        Ok((Some(status), _)) => {
+                            let protx: String = status
```

Comment:
protx hash can be empty (for example, for full nodes); maybe use node ID as a key?